### PR TITLE
32-run lint and test

### LINKED ISSUE #32: Add Lint, Format, Testing on CICD

### PR DESCRIPTION
**Pull request by rmc3408**
install checkout
node
dependencies
lint
test

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: FrontEnd CICD
+
+on: [pull_request]
+
+jobs:
+  run-checking:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checking version of Tools
```
Comment (sergiobtos) [inline .github/workflows/cicd.yaml]: what do you think about "Check Tool Versions"?

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: FrontEnd CICD
+
+on: [pull_request]
```
Comment (sergiobtos) [inline .github/workflows/cicd.yaml:3]: How about branches, is it necessary ? (source: **https://github.com/readme/guides/sothebys-github-actions**) 
```
on:
  pull_request:
    branches:
      - main  # Adjust this to your default branch
```

File: .github/workflows/cicd.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+      - name: Install dependencies
+        run: npm install
+      - name: Running typescript compiler, eslint and prettier
+        run: npm run lint
+      - name: Running Jest
+        run: npm run test
```
Comment (sergiobtos) [inline .github/workflows/cicd.yaml]: how about do the cache dependencies ? (source: **https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows**)
Reason: **Caching dependencies can significantly improve the performance of your CI/CD workflow, especially if your project has a large number of dependencies that take time to install. Caching helps avoid downloading and installing the same dependencies every time the workflow runs, which can save time and reduce the overall duration of your pipeline.**

```
- name: Cache Dependencies
        uses: actions/cache@v2
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-
```

File: .github/workflows/cicd.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  run-checking:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checking version of Tools
+        run: |
+          node -v
+          npm -v
+          aws --version
+      - name: Get repo
```
Comment (sergiobtos) [inline .github/workflows/cicd.yaml]: How about " **Checkout Repository** "?

File: .github/workflows/cicd.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        uses: actions/checkout@v3
+      - name: Use Node.js
```
Comment (sergiobtos) [inline .github/workflows/cicd.yaml]: How about "**Set Up Node.js**"?

**Review by sergiobtos — APPROVED** [review]
The YAML configuration is correct.  However, there are a few potential improvements and clarifications that you could consider. Just think and you can decide what is worth changing. Other than that looks good to me.

File: .github/workflows/cicd.yaml (hunk 2 above)
Comment (rmc3408) [reply .github/workflows/cicd.yaml:3]: Now, I focus to clean and test in all branches. Later i will do for specific branch, like building and deploying for main branch.

**Review by sergiobtos — APPROVED** [review]
LGTM

**Pull request merged**